### DISCUSSION TITLE
GitHub Pages - "commercial use"

### DISCUSSION THREAD

**Discussion by JOSCHLINER** [community/community#74742] [Other Feature Feedback, Questions, & Ideas] [answered]
According to Github's own documentation on GitHub Pages (<https://docs.github.com/en/pages/getting-started-with-github-pages/about-github-pages>): 
> GitHub Pages is not intended for or allowed to be used as a free web-hosting service to run your online business, e-commerce site, or any other website that is primarily directed at either facilitating commercial transactions or providing commercial software as a service (SaaS).

If the site does not have any transactions involved, being just a simple side for your local small business, would that breach Github's Guidelines? I would say no as it isn't an online business just a digital sign.
Please share your thoughts, though I would prefer getting an official answer as neither I nor GitHub wants me to breach ToS.

**Comment by BlackishGreen33** [comment, marked as answer]
Hello @JOSCHLINER

According to the information provided in the GitHub Pages documentation, GitHub Pages is not intended for hosting websites that are primarily directed at facilitating commercial transactions or providing commercial software as a service (SaaS). While it is clear that GitHub Pages is not meant for running an online business or e-commerce site, the specific guidelines regarding simple websites for local small businesses are not explicitly mentioned.

To get an official answer and clarify any doubts, it would be best to directly consult the GitHub Support or refer to the GitHub Terms of Service. They will be able to provide you with accurate information regarding the specific use case you have in mind and whether it complies with their guidelines. It's important to ensure compliance with the terms to avoid any potential violations.

Hope this helps.

**Comment by oseiskar** [comment]
It would indeed be great if Github could clatify their ToS regarging this.

In addition to the "Getting Started" docs, that the OP links to, the same vague statement also appears in the Terms of Service https://docs.github.com/en/site-policy/github-terms/github-terms-for-additional-products-and-features#pages:

> Each Account comes with access to the _GitHub Pages static hosting service_. GitHub Pages is intended to host static web pages, but primarily as a showcase for personal and organizational projects.
>
> GitHub Pages is not intended for or allowed to be used as a free web hosting service to run your online business, e-commerce site, or any other website that is primarily directed at either facilitating commercial transactions or providing commercial software as a service (SaaS). Some monetization efforts are permitted on Pages, such as donation buttons and crowdfunding links.

This kind of language is good for a Getting Started page but it is not appropriate for a legal document like the Terms of Service, which should clearly state what you can or cannot do, not what the service is "primarily intended for". It is very strange to see this in a ToS document. The way I read this is that:

 1. Sites like this are great and allowed: https://humanaigc.github.io/animate-anyone/
 2. Obvious spam and abusive content is banned (like on most platforms)
 3. Everything else is gray zone which is not "primarly intended" use, and you may or may not be violating the ToS

**Reply by wildone** [reply]
Did anything come of this?

**Reply by oseiskar** [reply]
The ToS is still the same (unclear). We have been using Github pages to host the documentation of our commercial SDK https://spectacularai.github.io/docs/sdk since 2022 and have not received any complaints from Github so far. However, I doubt the rules are strictly, actively or consistently enfoced for most smaller sites that fall withing the gray area.

**Reply by LiteBrite82** [reply]
Hi @oseiskar 

We have had similar questions before: https://github.com/orgs/community/discussions/126006
You are correct on the "grey area"; its hard to be specific with sooo many different applications out there. 

As long as the website and information contained in it is free and available to everyone; you should be ok.

**Reply by LiteBrite82** [reply]
https://humanaigc.github.io/animate-anyone/ is a great example of what is allowed. Open source collaboration is what we are all about. The catch is; if at some point in the future the app is released and becomes SaaS the gh-page may need to be reconsidered.

**Reply by mpfaff** [reply]
Hey @LiteBrite82

Would you be able to comment on the specific example JOSCHLINER gave?

> If the site does not have any transactions involved, being just a simple side for your local small business, would that breach Github's Guidelines?

If more specifics would be necessary to give a definitive answer, I can try to narrow it down for you.

**Reply by oseiskar** [reply]
Also @LiteBrite82 : where do you get this [definitive answers](https://github.com/orgs/community/discussions/126006#discussioncomment-9610748) from? Are they official and legally binding? If yes, could you ask that department to publish these guidelines on the ToS page, instead of drip feeding them in these chats. 

The current ToS is vague and unclear: a large fraction of current GH page uses are now in the "gray zone" and this is a bad thing. This should be fixed by clarifying the ToS, which is doable if there if MSFT/Github have motivation to do it instead of keeping it vague on purpose. It is not "hard to be specific with so many different applications". You do not need to list them all to clarify the rules, and Github probably employs quite a lot of people in legal/technical roles whose job includes outputting contractual language that categorizes websites and software in some reasonable way.

**Reply by LiteBrite82** [reply]
💬 Your Product Feedback Has Been Submitted 🎉
Thank you for taking the time to share your insights with us! Your feedback is invaluable as we build a better GitHub experience for all our users.

Here's what you can expect moving forward ⏩
- Your input will be carefully reviewed and cataloged by members of our product teams. 
- Other users may engage with your post, sharing their own perspectives or experiences. 
- GitHub staff may reach out for further clarification or insight. 

Where to look to see what's shipping 👀
- Read the [Changelog](https://github.blog/changelog/) for real-time updates on the latest GitHub features, enhancements, and calls for feedback.
- Explore our [Product Roadmap](https://github.com/orgs/github/projects/4247), which details upcoming major releases and initiatives.

What you can do in the meantime 💻
- Upvote and comment on other user feedback Discussions that resonate with you.
- Add more information at any point! Useful details include: use cases, relevant labels, desired outcomes, and any accompanying screenshots.

As a member of the GitHub community, your participation is essential. While we can't promise that every suggestion will be implemented, we want to emphasize that your feedback is instrumental in guiding our decisions and priorities.

Thank you once again for your contribution to making GitHub even better! We're grateful for your ongoing support and collaboration in shaping the future of our platform. ⭐

**Comment by Shxmeem** [comment]
Did you guys found any solution for this?
